tests/test_dialogs.py

```diff
@@ -0,0 +1,154 @@
+
+    assert not project_dialog.project_name_error.isHidden()
+    assert project_dialog.name_error
+    assert not project_dialog.project_folder_error.isHidden()
+    assert project_dialog.folder_error
+
+    # tests the project name line edit error displayed.
+    project_dialog.project_folder.setText("test-folder")
+    QtTest.QTest.mouseClick(project_dialog.create_button, QtCore.Qt.MouseButton.LeftButton)
+
+    assert not project_dialog.project_name_error.isHidden()
+    assert project_dialog.name_error
+    assert project_dialog.project_folder_error.isHidden()
+    assert not project_dialog.folder_error
+
+    # tests the project folder line edit error displayed.
+    project_dialog.project_name.setText("test-name")
+    project_dialog.project_folder.setText("")
+    QtTest.QTest.mouseClick(project_dialog.create_button, QtCore.Qt.MouseButton.LeftButton)
+
+    assert project_dialog.project_name_error.isHidden()
+    assert not project_dialog.name_error
+    assert not project_dialog.project_folder_error.isHidden()
+    assert project_dialog.folder_error
+
+
+@patch.object(MockParentWindow, "createNewProject")
+def test_create_button(mock_create_new_project, setup_project_dialog_widget):
+    """
+    Tests create button on the ProjectDialog class.
+    """
+    project_dialog, _ = setup_project_dialog_widget
+
+    project_dialog.project_name.setText("test-name")
+    project_dialog.project_folder.setText("test-folder")
+    QtTest.QTest.mouseClick(project_dialog.create_button, QtCore.Qt.MouseButton.LeftButton)
+    mock_create_new_project.assert_called_once()
+
+
+@patch.object(MockParentWindow, "toggleView")
+def test_cancel_button(mock_toggle_view, setup_project_dialog_widget):
+    """
+    Tests cancel button on the ProjectDialog class.
+    """
+    project_dialog, _ = setup_project_dialog_widget
+
+    project_dialog.project_name.setText("test-name")
+    project_dialog.name_error = True
+    project_dialog.project_folder.setText("test-folder")
+    project_dialog.folder_error = True
+
+    QtTest.QTest.mouseClick(project_dialog.cancel_button, QtCore.Qt.MouseButton.LeftButton)
+
+    mock_toggle_view.assert_called_once()
+
+    assert project_dialog.project_name.text() == ""
+    assert not project_dialog.name_error
+    assert project_dialog.project_folder.text() == ""
+    assert not project_dialog.folder_error
+
+
+@patch("os.listdir")
+@patch("PyQt6.QtWidgets.QFileDialog.getExistingDirectory")
+def test_browse_button(mock_get_existing_directory, mock_listdir, setup_project_dialog_widget):
+    """
+    Tests the browse button on the ProjectDialog class.
+    """
+    project_dialog, _ = setup_project_dialog_widget
+
+    # When empty folder is selected.
+    mock_get_existing_directory.return_value = "/test/folder/path"
+    mock_listdir.return_value = [".hiddenfile"]
+
+    QtTest.QTest.mouseClick(project_dialog.browse_button, QtCore.Qt.MouseButton.LeftButton)
```

Comment:
Please replace with  `QtTest.QTest.mouseClick(project_dialog.browse_button, QtCore.Qt.MouseButton.LeftButton)` and similar with `project_dialog.browse_button.click()`